optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-backend/src/main/java/org/optaplanner/workbench/screens/solver/backend/server/SolverValidator.java

```diff
@@ -131,7 +132,7 @@ private ValidationMessage createSolverFactory( final Path resourcePath,
                                                    final KieProject kieWorkbenchProject,
                                                    final boolean runSolver ) {
         final Builder builder = builderCache.assertBuilder( kieWorkbenchProject );
-        final InternalKieModule kieModule = (InternalKieModule) builder.getKieModule();
+        final InternalKieModule kieModule = (InternalKieModule) builder.getKieModuleIgnoringErrors();
```

Comment:
@mcimbora @ge0ffrey We had to change the "generic validator" mechanism as the previous changes broke XLS Decision Table validation. The revised changes (see [here](https://github.com/droolsjbpm/kie-wb-common/pull/479)) "build" the ```Builder``` (if not already built) in the ```builderCache``` that you are also using here. 

Your ```SolverValidatorTest.projectBuildError()``` test failed with these changes as ```SolverValidator.createSolverFactory()``` gets a ```KieModule``` from ```Builder``` (which is now "built" for validation) which includes compilation errors. Using ```builder.getKieModuleIgnoringErrors()``` ensures you can retrieve a ```KieModule``` even if the module contains compilation errors.